wrapper/src/main/java/software/amazon/jdbc/plugin/OpenedConnectionTracker.java

```diff
@@ -134,8 +140,11 @@ public void invalidateCurrentConnection(final HostSpec hostSpec, final Connectio
     }
 
     final Queue<WeakReference<Connection>> connectionQueue = openedConnections.get(host);
-    logConnectionQueue(host, connectionQueue);
-    connectionQueue.removeIf(connectionWeakReference -> Objects.equals(connectionWeakReference.get(), connection));
+    if (connectionQueue != null) {
+      logConnectionQueue(host, connectionQueue);
+      connectionQueue.removeIf(connectionWeakReference -> connectionWeakReference != null
+          && Objects.equals(connectionWeakReference.get(), connection));
```

Comment:
Should we also remove if connectionWeakReference is null? 
```suggestion
      connectionQueue.removeIf(connectionWeakReference -> connectionWeakReference == null || (connectionWeakReference != null
          && Objects.equals(connectionWeakReference.get(), connection)));
```